.github/workflows/stage-lint.yml

```
@@ -23,6 +23,13 @@ jobs:
         uses: golangci/golangci-lint-action@v3
         with:
           version: v1.46.2
+      - run: go mod tidy
+      - name: Undo go mod tidy changes if successful
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+            git checkout ./go.mod && git checkout ./go.sum
```

I think this would cause the check to exit with an error code and a helpful log message if `go.sum` isn't clean:

```suggestion
            echo "::error go.mod not tidy"
            exit 1
```